bin/i18n/sync-codeorg-in.rb

```diff
@@ -67,18 +66,100 @@ def localize_dsl_content
+        url = get_level_url_key(script, level)
+        if level.is_a?(DSLDefined)
+          level.contained_levels.each do |contained_level|
+            dsl_i18n_strings.deep_merge! get_dsl_i18n_strings(contained_level)
+          end
+
+          dsl_i18n_strings.deep_merge! get_dsl_i18n_strings(level)
+        else
+          # display_name
+          if level.display_name
+            script_strings['display_name'][url] = level.display_name
+          end
+          # short_instructions
+          if level.short_instructions
+            script_strings['short_instructions'][url] = level.short_instructions
+          end
+          # long_instructions
+          if level.long_instructions
+            script_strings['long_instructions'][url] = level.long_instructions
+          end
+
+          # failure_message_overrides
+          if level.class <= Blockly && level.failure_message_override
+            script_strings['failure_message_overrides'][url] = level.failure_message_override
+          end
+
+          # authored_hints
+          if level.authored_hints
+            authored_hints = JSON.parse(level.authored_hints)
+            script_strings['authored_hints'][url] = Hash.new unless authored_hints.empty?
+            authored_hints.each do |hint|
+              script_strings['authored_hints'][url][hint['hint_id']] = hint['hint_markdown']
+            end
+          end
+          # callouts
+          if level.callout_json
+            callouts = JSON.parse(level.callout_json)
+            script_strings['callouts'][url] = Hash.new unless callouts.empty?
+            callouts.each do |callout|
+              script_strings['callouts'][url][callout['localization_key']] = callout['callout_text']
+            end
+          end
+
+          level_xml = Nokogiri::XML(level.to_xml, &:noblanks)
+          blocks = level_xml.xpath('//blocks').first
+          if blocks
+            ## Categories
+            block_categories = blocks.xpath('//category')
+            script_strings['block_categories'][url] = Hash.new unless block_categories.empty?
+            block_categories.each do |category|
+              name = category.attr('name')
+              script_strings['block_categories'][url][name] = name if name
+            end
+
+            ## Function Names
+            functions = blocks.xpath("//block[@type=\"procedures_defnoreturn\"]")
+            script_strings['function_names'][url] = Hash.new unless functions.empty?
+            functions.each do |function|
+              name = function.at_xpath('./title[@name="NAME"]')
+              script_strings['function_names'][url][name.content] = name.content if name
+            end
+          end
+
         end
+      end
 
-        i18n_strings.deep_merge! get_dsl_i18n_strings(level)
+      script_strings.delete_if {|_key, value| value.nil? || value.try(:empty)}
+
+      script_i18n_directory = "../i18n/locales/source/course_content"
+      if script.version_year
```

Comment:
oh, this requirement might have gotten lost along the way, but this should be something like

```ruby
if script.version_year
  script_i18n_directory = "#{script_i18n_directory}/#{script.version_year}"
elsif ScriptConstants.script_in_category?(:hoc, script)
  script_i18n_directory = "#{script_i18n_directory}/Hour of Code"
else
  script_i18n_directory = "#{script_i18n_directory}/other"
end
```

Reply:
Done.